app/assets/src/components/views/SampleViewV2/ReportTable.jsx

```diff
@@ -354,10 +390,36 @@ class ReportTable extends React.Component {
   };
 
   toggleExpandGenus = ({ taxId }) => {
-    const { expandedGenusIds } = this.state;
+    const { data } = this.props;
+    const { expandedGenusIds, expandAllOpened } = this.state;
     expandedGenusIds.delete(taxId) || expandedGenusIds.add(taxId);
 
-    this.setState({ expandedGenusIds: new Set(expandedGenusIds) });
+    const newExpandedAllOpened =
+      expandedGenusIds.size === data.length ||
+      (!!expandedGenusIds.size && expandAllOpened);
+
+    this.setState({
+      expandedGenusIds: new Set(expandedGenusIds),
+      expandAllOpened: newExpandedAllOpened,
+    });
+  };
+
+  toggleExpandAll = () => {
+    const { data } = this.props;
+    const { expandAllOpened } = this.state;
+
+    if (expandAllOpened) {
+      this.setState({
+        expandedGenusIds: new Set(),
+        expandAllOpened: false,
+      });
+    } else {
+      this.setState({
+        expandedGenusIds: new Set(map("taxId", data)),
+        expandAllOpened: true,
+      });
+    }
+    event.stopPropagation();
```

Comment:
Can you add a comment around why `event.stopPropagation` is necessary? These calls can be hard to understand so it's good to explain them.

Reply:
Actually it is not even needed with the final solution of having a dedicated column for this. Removed.